### ISSUE TITLE
Add custom html content in Dots?

### ISSUE THREAD

**Issue by thakurpunk** [OwlCarousel2/OwlCarousel2#1350]
Hello!! Is there any way to put some custom html content in dots so that we can put some images and text in to make it something like thumbnails slider? If yes then please tell me how.

Thanks in advance :)

**Comment by DmitriyRF** [comment]
+1

**Comment by eparisio** [comment]
with dotsData you can add custom html to the dots

**Comment by tonykiefer** [comment]
I think this is the solution you are looking for.  http://vertx.com/edc-transit-sling

If it is...let me know and I'll explain to you how I did it...This is a dual synced carousel.  In this example I use nav icons, but in another solution is use dots instead.  But I think the dual synced carousel is what you need

**Comment by michellomp** [comment]
@tonykiefer can you explain how do you do that i thought synced where depreciated in the new owl carousel 2

**Comment by thakurpunk** [comment]
@eparisio Are you sure this would work?

**Comment by thakurpunk** [comment]
@tonykiefer Yes this is the same I am asking for. But can we have the same with videos and thumbnails? And thumbnails with a caption text on hover.

**Comment by eparisio** [comment]
@thakurpunk for sure the code is present in [owl.navigation.js](https://github.com/OwlCarousel2/OwlCarousel2/blob/develop/src/js/owl.navigation.js)

sorry but i mada an error on the previous code you need to set it in html:

`$(elem).owlCarousel({
dotsData: true
});

\<ul class="owl-carousel"\\>
\<li class="carousel-slot" data-dot="\<p\>text 1\</p\>" \>slide 1\</li\>
\<li class="carousel-slot" data-dot="\<p\>text 2\</p\>" \>slide 2\</li\>
\<li class="carousel-slot" data-dot="\<p\>text 3\</p\>" \>slide 3\</li\>
\</ul>
`

there an example [fiddle](https://jsfiddle.net/4xymnwey/)

they remove the demo and example from documentation but it's implemented

there is the param [line 141](https://github.com/OwlCarousel2/OwlCarousel2/blob/develop/src/js/owl.navigation.js#L141), and the initialize at [line 76](https://github.com/OwlCarousel2/OwlCarousel2/blob/develop/src/js/owl.navigation.js#L76)

**Comment by thakurpunk** [comment]
@eparisio this is fine. but what we can do if we want some html content like buttons and text altogether? This one id good for simpler text.

**Comment by eparisio** [comment]
you can add any html inside, you are not restricted to text or paragraph.

you can do

> \<li class="carousel-slot" data-dot="\<div\>\<button class='btn'\>button\</button\>\<p\>text 1\</p\>\</div\>" \>slide 1\</li\>

**Comment by thakurpunk** [comment]
@eparisio have you done something like this before? can I see some example

**Comment by eparisio** [comment]
there is the fiddle in the example before, you can play with it!

**Comment by luizfilipe2557** [comment]
@thakurpunk, f you have some time, take a look on: #1377

**Comment by thakurpunk** [comment]
thanks @eparisio :)

**Comment by luizfilipe2557** [comment]
@thakurpunk at this moment I'm writing a plugin to sync the carousels until nobody add this funcionality to Owl. I will post the source code soon.

**Comment by tonykiefer** [comment]
Most recently someone posted the way they sync two carousels...take a look here

https://github.com/OwlCarousel2/OwlCarousel2/issues/1377#issuecomment-216883486

**Comment by tonykiefer** [comment]
@thakurpunk - until they add the dual synced carousels I think they just expect us to customize the code ourselves and help each other.  So far so good.  This is a good community with a lot of great ideas.

**Comment by tonykiefer** [comment]
@thakurpunk please see the attached.  For styling...you'll need to write your own.

@luizfilipe2557 it looks like we both had the same resources...nice work on your code...feel free to review mine...also a live version...http://vertx.com/edc-transit-sling

I'll be launching another site where I use dots for the nav...I'll be posting that come the 18th

[Dual Sync Carousel.txt](https://github.com/OwlCarousel2/OwlCarousel2/files/249407/Dual.Sync.Carousel.txt)

**Comment by luizfilipe2557** [comment]
Hey everyone, plugin to sync the carousels is ready. Take a look on Issue #1377 in [this comment](https://github.com/OwlCarousel2/OwlCarousel2/issues/1377#issuecomment-216992946).

**Comment by mdeboer** [comment]
@luizfilipe2557 works great, thanks 👍  

Only bug I still have (with my own syncing code and with your plugin) that the index is completely off between sliders when set to loop. Still not sure wether it is due me cloning the slider element (before initializing owl) or it's really a bug somewhere. All works fine without looping!

Both the event's item.index as jQuery(this).index() are both off (and both the same). Somewhere the indexes get messed up.

**Comment by luizfilipe2557** [comment]
@mdeboer I found the problem that generates the bug. I will be working on it soon, my first idea is to set data-index on the slides that represents the exhibition index and order when the carousel is in loop and use this data-index instead of element index. I will create a branch for it and describe the problem so that users like you and me can work on it.

Thanks for the feedback, really appreciate it. :)

**Comment by mdeboer** [comment]
@luizfilipe2557 that was my idea too but for some reason that didn't work too well. At least not when setting it before initializing owl. Haven't tried for very long though, haven't got much time for this project and worked my way around this time.

Yet, a solution would be very nice! 👍

**Comment by luizfilipe2557** [comment]
@mdeboer I don't have much time too, but together with git we can make it by small steps, even because the core of the plugin is ready! :smile:

**Comment by daviddeutsch** [comment]
As noted in the other thread - there is a core plugin for this (linking owl instances) in the works.

**Comment by thakurpunk** [comment]
@tonykiefer and @luizfilipe2557 Thanks, I have created one using suggested codes, you can see it http://mylkshake.co/mylshaketv/ just review and do let me know how is it.

**Comment by luizfilipe2557** [comment]
Hllo guys, as I said in #1377 thanks to @eparisio we could get this in [version two](https://github.com/luizfilipe2557/owlcarousel-sync-plugin/tree/version-2) (https://github.com/luizfilipe2557/owlcarousel-sync-plugin/tree/version-2), we made the plugin as a "native plugin", I would be very happy if you all join us in this development. :smile: 
<hr>
- [Sync Plugin: Master Branch (V1)](https://github.com/luizfilipe2557/owlcarousel-sync-plugin/)
- [Sync Plugin: Version 2 Branch (V2)](https://github.com/luizfilipe2557/owlcarousel-sync-plugin/tree/version-2)
- [Sync Plugin: V2.2 Plugin example 1 (jsFiddle)](https://fiddle.jshell.net/odahcam/310d470p/)
- [Sync Plugin: V2.2 Plugin example 2 (jsFiddle)](https://fiddle.jshell.net/odahcam/kdc874zm/)

**Comment by thakurpunk** [comment]
@luizfilipe2557 have you reviewed mine?

**Comment by luizfilipe2557** [comment]
@thakurpunk Yeah! Your code is very very close to my **v1** code. Just like me and the people who helped me you don't have loop:true feature too, what makes it closer to our **stable** code. Feel free to work on our code too!
<hr>
Take a look: https://github.com/luizfilipe2557/owlcarousel-sync-plugin/blob/master/owl.sync.js
Take a look on **V2** too :smile:  : https://github.com/luizfilipe2557/owlcarousel-sync-plugin/blob/version-2/owl.sync.js

**Comment by thakurpunk** [comment]
thanks @luizfilipe2557

**Comment by mitkovalkov** [comment]
Hi Guys, 
when I use dotsData: true in my slider and put a link in data-dote="<a href="https://github.com/">", the link does not work correctly, any suggestions?

**Comment by odahcam** [comment]
@mitkovalkov Sorry, but here is not the place for this kind of question.

**But, ** please post a question at stackoverflow.com with an example and I'll take a look. :)

**Comment by mitkovalkov** [comment]
@odahcam Thanks for the advice, I already solved the problem.
